codegen/example/example_server.go

```diff
@@ -274,11 +275,15 @@ func main() {
 				u.Host = *domainF
 			}
 			if *{{ $u.Transport.Type }}PortF != "" {
-				h := strings.Split(u.Host, ":")[0]
-				u.Host = h + ":" + *{{ $u.Transport.Type }}PortF
+				h, _, err := net.SplitHostPort(u.Host)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "invalid URL %#v: %s\n", u.Host, err)
+					os.Exit(1)
+				}
+				u.Host = net.JoinHostPort(h, *{{ $u.Transport.Type }}PortF)
 			} else if u.Port() == "" {
-				u.Host += ":{{ $u.Port }}"
-			}
+				u.Host = net.JoinHostPort(u.Host, ":{{ $u.Port }}")
+			}		
```

Comment:
Nitpick: please remove the extra whitespaces

Reply:
done